server/src/main/java/io/crate/metadata/doc/DocTableInfoFactory.java

```diff
@@ -219,42 +214,26 @@ public DocTableInfo create(IndexMetadata indexMetadata) {
 
     /**
      * Deprecated. Only used for upgrading old templates to new RelationMetadata.
+     *
+     * Note that the {@link DocTableInfo#versionCreated()} of the returned instance could be wrong.
+     * See https://github.com/crate/crate/pull/17178
      */
     @Deprecated
-    public DocTableInfo create(IndexTemplateMetadata indexTemplateMetadata, Metadata metadata) {
-        RelationName relationName = RelationName.fromIndexName(indexTemplateMetadata.name());
-        Map<String, Object> mappingSource = XContentHelper.toMap(
-            indexTemplateMetadata.mapping().compressedReference(),
-            XContentType.JSON
-        );
+    public DocTableInfo create(IndexTemplateMetadata template, Metadata metadata) {
+        RelationName relationName = RelationName.fromIndexName(template.name());
+        Map<String, Object> mappingSource = IndexTemplateUpgrader.updateMapping(template.mapping());
         mappingSource = Maps.getOrDefault(mappingSource, "default", mappingSource);
-        Settings tableParameters = indexTemplateMetadata.settings();
-        Version versionCreated = tableParameters.getAsVersion(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT);
```

Comment:
Removed this completely here because this variant of `create` is only called by the `MetadataUpgradeService` which has its own logic to fix the version - including an additional oid check which was missing here.